unstdlib/standard/list_.py

```diff
@@ -145,6 +145,32 @@ def iterate_flatten(q):
 
     return chain.from_iterable(q)
 
+def partition(iterable, predicate=bool):
```

Comment:
How do I get the non-matching items then?

Here's an example of how I use it in actual code:
```
     published, drafts = partition(articles, lambda article: article.is_published())
     ...
     render(...
         published=itertools.islice(published, 20),
         drafts=itertools.islice(drafts, 20),
     )
     
```

With ```ifilter``` I'd have to iterate the original list and evaluate the predicate twice.